jplag/src/main/java/de/jplag/reportingV2/reportobject/ReportObjectFactory.java

```diff
@@ -0,0 +1,209 @@
+
+/**
+ * Factory class, responsible for converting a JPlagResult object to Overview and Comparison DTO classes.
+ */
+public class ReportObjectFactory {
+
+    /**
+     * Converts a JPlagResult to a JPlagReport.
+     * @return JPlagReport for the given JPlagResult.
+     */
+    public static JPlagReport getReportObject(JPlagResult result) {
+        OverviewReport overviewReport = generateOverviewReport(result);
+        List<ComparisonReport> comparisons = generateComparisonReports(result);
+        return new JPlagReport(overviewReport, comparisons);
+    }
+
+    /**
+     * Generates an Overview DTO of a JPlagResult.
+     */
+    private static OverviewReport generateOverviewReport(JPlagResult result) {
+        List<JPlagComparison> comparisons = result.getComparisons();
+        OverviewReport overviewReport = new OverviewReport();
+
+        // New update in JPlagOption. Not a single root dir but a list. Needs re-thinking and
+        // implementation also in report model.
+        overviewReport.setSubmission_folder_path(result.getOptions().getRootDirectoryNames().get(0));
+
+        String baseCodePath = result.getOptions().hasBaseCode() ? result.getOptions().getBaseCodeSubmissionName().orElse("") : "";
+        overviewReport.setBase_code_folder_path(baseCodePath);
+
+        overviewReport.setLanguage(result.getOptions().getLanguage().getName());
+        overviewReport.setFile_extensions(List.of(result.getOptions().getFileSuffixes()));
+        overviewReport.setSubmission_ids(extractSubmissionNames(comparisons));
+        overviewReport.setFailed_submission_names(List.of());  // No number of failed submissions
+        overviewReport.setExcluded_files(getExcludedFilesNames(result.getOptions())); // Read exclusion file
+        overviewReport.setMatch_sensitivity(result.getOptions().getMinimumTokenMatch());
+        overviewReport.setDate_of_execution(getDate());
+        overviewReport.setExecution_time(result.getDuration());
+        overviewReport.setComparison_names(getComparisonNames(comparisons));
+        overviewReport.setMetrics(getMetrics(result));
+        overviewReport.setClusters(getClusters(result));
+
+        return overviewReport;
+    }
+
+    /**
+     * Generates detailed ComparisonReport DTO for each comparison in a JPlagResult.
+     * @return A list with ComparisonReport DTOs.
+     */
+    private static List<ComparisonReport> generateComparisonReports(JPlagResult result) {
+        List<ComparisonReport> comparisons = new ArrayList<>();
+        result.getComparisons().forEach(c -> comparisons.add(new ComparisonReport(c.getFirstSubmission().getName(), c.getSecondSubmission().getName(),
+                c.similarity(), getFilesForSubmission(c.getFirstSubmission()), getFilesForSubmission(c.getSecondSubmission()), c.getMatches().stream()
+                        .map(m -> convertMatchToReportMatch(c, m, result.getOptions().getLanguage().usesIndex())).collect(Collectors.toList()))));
```

Comment:
I would suggest to replace the lambdas by non-lambda functions/methods .. that may be more readable :)